public/app/features/datasources/api.ts

```diff
@@ -68,7 +68,13 @@ export const createDataSource = (dataSource: Partial<DataSourceSettings>) =>
 
 export const getDataSourcePlugins = () => getBackendSrv().get('/api/plugins', { enabled: 1, type: 'datasource' });
 
-export const updateDataSource = (dataSource: DataSourceSettings) =>
-  getBackendSrv().put(`/api/datasources/uid/${dataSource.uid}`, dataSource);
+export const updateDataSource = (dataSource: DataSourceSettings) => {
+  // we're setting showErrorAlert and showSuccessAlert to false to suppress the popover notifications. Request result will now be
+  // handled by the data source config page
+  return getBackendSrv().put(`/api/datasources/uid/${dataSource.uid}`, dataSource, {
+    showErrorAlert: false,
+    showSuccessAlert: false,
+  });
+};
```

Comment:
I am not seeing a try-catch where updateDataSource is called that handles the error

Reply:
@torkelo regarding

>Add some proper typing to DataSourceApi.testDatasource: Promise<any> (a Promise that returns an object with a status: error | string, and message)

you mean adding a return type here massaging the data that comes from the `put` call, right ?